tcms/xmlrpc/api/testcase.py

```diff
@@ -1246,13 +1243,10 @@ def notification_remove_cc(request, case_ids, cc_list):
 
     try:
         tc_ids = pre_process_ids(case_ids)
-        cursor = connection.writer_cursor
-        ids_values = ",".join(itertools.repeat('%s', len(tc_ids)))
-        email_values = ",".join(itertools.repeat('%s', len(cc_list)))
-        sql = TC_REMOVE_CC % (ids_values, email_values)
-        tc_ids.extend(cc_list)
-        cursor.execute(sql, tc_ids)
-        transaction.commit_unless_managed()
+
+        for tc in TestCase.objects.filter(pk__in=tc_ids).iterator():
```

Comment:
Here, the queried data size can be reduced by

```
for tc in TestCase.objects.filter(pk__in=tc_ids).only('pk').iterator():
```

Removing CC here only requires the relationship between ``TestCase`` and ``TestCaseEmailSettings``, no other ``TestCase`` attributes are needed.